mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb

```diff
@@ -4734,3 +4734,102 @@ resource "google_container_node_pool" "primary_nodes" {
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccContainerNodePool_privateRegistryEnabled(cluster, nodepool, networkName, subnetworkName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"google_container_node_pool.np",
+						"node_config.0.containerd_config.0.private_registry_access_config.0.enabled",
+						"true",
+					),
+					resource.TestCheckResourceAttr(
+						"google_container_node_pool.np",
+						"node_config.0.containerd_config.0.private_registry_access_config.0.certificate_authority_domain_config.#",
+						"2",
+					),
+					// First CA config
+					resource.TestCheckResourceAttr(
+						"google_container_node_pool.np",
+						"node_config.0.containerd_config.0.private_registry_access_config.0.certificate_authority_domain_config.0.fqdns.0",
+						"my.custom.domain",
+					),
+					resource.TestCheckResourceAttr(
+						"google_container_node_pool.np",
+						"node_config.0.containerd_config.0.private_registry_access_config.0.certificate_authority_domain_config.0.gcp_secret_manager_certificate_config.0.secret_uri",
+						"projects/0000000000/secrets/my-cert/versions/1",
+					),
+					// Second CA config
+					resource.TestCheckResourceAttr(
+						"google_container_node_pool.np",
+						"node_config.0.containerd_config.0.private_registry_access_config.0.certificate_authority_domain_config.1.fqdns.0",
+						"10.1.2.32",
+					),
+					resource.TestCheckResourceAttr(
+						"google_container_node_pool.np",
+						"node_config.0.containerd_config.0.private_registry_access_config.0.certificate_authority_domain_config.1.gcp_secret_manager_certificate_config.0.secret_uri",
+						"projects/0000000000/secrets/my-other-cert/versions/latest",
+					),
+				),
+			},
+		},
+	})
+}
+
+func testAccContainerNodePool_privateRegistryEnabled(cluster, nodepool, network, subnetwork string) string {
+	return fmt.Sprintf(`
+resource "google_container_cluster" "cluster" {
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  deletion_protection = false
+  network    = "%s"
+  subnetwork    = "%s"
+}
+	
+resource "google_container_node_pool" "np" {
+  name               = "%s"
+  location           = "us-central1-a"
+  cluster            = google_container_cluster.cluster.name
+  initial_node_count = 1
+	
+  node_config {
+	oauth_scopes = [
+      "https://www.googleapis.com/auth/cloud-platform",
+    ]
+    machine_type = "n1-standard-8"
+    image_type = "COS_CONTAINERD"
+    containerd_config {
+      private_registry_access_config {
+        enabled = true
+        certificate_authority_domain_config {
+          fqdns = [ "my.custom.domain", "10.0.0.127:8888" ]
+          gcp_secret_manager_certificate_config {
+            secret_uri = "projects/0000000000/secrets/my-cert/versions/1"
```

Comment:
Also, could these tests provision these secret resources so that a dummy value isn't used here? Even if the tests pass currently there's a chance the API may change and begin validating inputs like this.

Reply:
This is the definition in proto file:

```
message [CertificateAuthorityDomainConfig](https://source.corp.google.com/piper///depot/google3/google/container/v1/cluster_service.proto;bpv=1;bpt=1;l=1600?q=cluster_service.proto&gsn=CertificateAuthorityDomainConfig&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fgoogle3%3Flang%3Dprotobuf%3Fpath%3Dgoogle%2Fcontainer%2Fv1%2Fcluster_service.proto%234.16.3.0.3.0) {
      // GCPSecretManagerCertificateConfig configures a secret from
      // [Google Secret Manager](https://cloud.google.com/secret-manager).
      message [GCPSecretManagerCertificateConfig](https://source.corp.google.com/piper///depot/google3/google/container/v1/cluster_service.proto;bpv=1;bpt=1;l=1603?q=cluster_service.proto&gsn=GCPSecretManagerCertificateConfig&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fgoogle3%3Flang%3Dprotobuf%3Fpath%3Dgoogle%2Fcontainer%2Fv1%2Fcluster_service.proto%234.16.3.0.3.0.3.0) {
        // Secret URI, in the form
        // "projects/$PROJECT_ID/secrets/$SECRET_NAME/versions/$VERSION".
        // Version can be fixed (e.g. "2") or "latest"
        string [secret_uri](https://source.corp.google.com/piper///depot/google3/google/container/v1/cluster_service.proto;bpv=1;bpt=1;l=1607?q=cluster_service.proto&gsn=secret_uri&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fgoogle3%3Flang%3Dprotobuf%3Fpath%3Dgoogle%2Fcontainer%2Fv1%2Fcluster_service.proto%234.16.3.0.3.0.3.0.2.0) = 1 [(datapol.semantic_type) = ST_USER_CONTENT];
      }
```

I am not sure if we can we provision these secret resources.

Reply:
Here's an example of how the test step can include secrets to support testing a given resource. The secret version's data doesn't matter, but the correct IAM permissions will need to be in place: 

https://github.com/GoogleCloudPlatform/magic-modules/blob/53a7d7ffd2f07900f15ceaf6284b5a3ed761f823/mmv1/third_party/terraform/services/integrationconnectors/resource_integration_connectors_connection_test.go#L113-L170

Reply:
Added secrets as the resource.